### PR TITLE
Increase the default TF-M crypto partition stack size when 4096 bit RSA keys are enabled

### PR DESCRIPTION
**Pull request by vili-nordic**

### CODE REVIEW

File: samples/crypto/rsa/src/key.h [hunk 1]
```diff
@@ -0,0 +1,207 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
```
Comment (mswarowsky) [inline samples/crypto/rsa/src/key.h:2]:
```suggestion
 * Copyright (c) 2024 Nordic Semiconductor ASA
```

**Review by mswarowsky — APPROVED** [review]
One small nit, but rest lgtm

File: samples/crypto/rsa/README.rst [hunk 2]
```diff
@@ -26,13 +26,13 @@ The sample performs the following operations:
 1. Initialization:
 
    a. The Platform Security Architecture (PSA) API is initialized.
-   #. A random RSA key pair is generated and imported into the PSA crypto keystore.
-   #. The public key of the RSA key pair is imported into the PSA crypto keystore.
+   #. A RSA key pair is imported into the PSA crypto keystore.
```
Comment (frkv) [inline samples/crypto/rsa/README.rst:29]: It is unfortunate that we have to rely on a pre-generated key, but I think it should be mentioned that this key is not usable in any end-products. Maybe this text should be given in italics

```suggestion
   #. A RSA key pair is imported into the PSA crypto keystore.
   #. *Note that this is a test key and it should never be used in a product!*
```

**Review by SebastianBoe — COMMENTED** [review]
The title might mislead a customer into thinking using 4096 bit keys was not supported with TF-M previously.

The customer could have 4096 bit keys in his TF-M application by doing

CONFIG_TFM_CRYPTO_PARTITION_STACK_SIZE=0x4000

maybe:

Increase the default TF-M crypto partition stack size when 4096 bit RSA keys are enabled

**Review by SebastianBoe — APPROVED** [review]

File: samples/crypto/rsa/README.rst [hunk 3]
```diff
@@ -26,13 +26,15 @@ The sample performs the following operations:
...
```
Comment (mia-ko) [inline samples/crypto/rsa/README.rst:29]:
```suggestion
   #. An RSA key pair is imported into the PSA crypto keystore.
```

File: samples/crypto/rsa/README.rst [hunk 4]
```diff
@@ -26,13 +26,15 @@ The sample performs the following operations:
...
+   #. The public key is imported into the PSA crypto keystore.
+
+   *Note that the keys used in the sample are for demonstration purposes only and never be used in a product.*
```
Comment (mia-ko) [inline samples/crypto/rsa/README.rst:32]:
```suggestion
   #. The public key is imported into the PSA crypto keystore.

      .. note::
         The keys used in the sample are for demonstration purposes only and must never be used in a product.
         ```

**Review by mia-ko — CHANGES_REQUESTED** [review]

File: samples/crypto/rsa/README.rst [hunk 5]
```diff
@@ -26,13 +26,17 @@ The sample performs the following operations:
...
 
    a. The Platform Security Architecture (PSA) API is initialized.
-   #. A random RSA key pair is generated and imported into the PSA crypto keystore.
-   #. The public key of the RSA key pair is imported into the PSA crypto keystore.
+   #. An RSA key pair is imported into the PSA crypto keystore.
+   #. The public key is imported into the PSA crypto keystore.
+
+      .. note::
+         The keys used in the sample are for demonstration purposes only and must never be used in a product.
+         ```
```
Comment (mia-ko) [inline samples/crypto/rsa/README.rst:34]: These sneaked in, you probably want to remove them :)

**Review by mia-ko — COMMENTED** [review]
Do these changes require a changelog update?

**Comment by vili-nordic** [comment]
I don't think updating the default stack-size for tf-m require changelog entry. The sample is showing something slightly different now (but probably a more "common" case), but I'm not sure how relevant that is for a changelog.

**Pull request merged**